File: data/code_pairs/fortran/793.f90
! Main Program
program main
    implicit none
    integer :: status

    status = run_tests()
    if (status == 0) then
        print *, "Fortran: Test Passed"
    else
        print *, "Fortran: Test Failed with status: ", status
    endif
end program main

! Function to perform tests based on the initial problem statement
function run_tests() result(status)
    implicit none
    integer :: status
    integer :: i
    integer, parameter :: m(4,4) = reshape([ (i, i=1, 16) ], [4, 4])
    integer, parameter :: sevens = count(m == 7)
    integer, parameter :: odd(4) = count(mod(m, 2) == 1, dim=1)
    integer, parameter :: even = count(mod(m, 2) == 0)

    status = 0
    if (sevens /= 1) then
        status = 1
    else if (any(odd /= [2,2,2,2])) then
        status = 2
    else if (even /= 8) then
        status = 3
    endif
end function run_tests

File: data/code_pairs/cpp/793.cpp
#include <iostream>
#include <array>
#include <numeric>
#include <algorithm>

int runTests();

int main() {
    int status = runTests();
    if (status == 0) {
        std::cout << "C++: Test Passed" << std::endl;
    } else {
        std::cout << "C++: Test Failed with status: " << status << std::endl;
    }
    return 0;
}

int runTests() {
    std::array<std::array<int, 4>, 4> m = {{{1, 2, 3, 4},
                                            {5, 6, 7, 8},
                                            {9, 10, 11, 12},
                                            {13, 14, 15, 16}}};
    int sevens = std::accumulate(m.begin(), m.end(), 0, [](int acc, const std::array<int, 4>& row) {
        return acc + std::count(row.begin(), row.end(), 7);
    });

    std::array<int, 4> odd = {};
    for (int i = 0; i < 4; ++i) {
        odd[i] = std::count_if(m[i].begin(), m[i].end(), [](int x) { return x % 2 == 1; });
    }

    int even = std::accumulate(m.begin(), m.end(), 0, [](int acc, const std::array<int, 4>& row) {
        return acc + std::count_if(row.begin(), row.end(), [](int x) { return x % 2 == 0; });
    });

    if (sevens != 1) return 1;
    if (!std::equal(odd.begin(), odd.end(), std::array<int, 4>{2, 2, 2, 2}.begin())) return 2;
    if (even != 8) return 3;

    return 0; // Test Passed
}